iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -610,4 +610,48 @@ cxx::error<TimerError> Timer::createErrorFromErrno(const int32_t errnum) noexcep
 }
 
 } // namespace posix
+
+namespace clock
+{
+DeadlineTimer::DeadlineTimer(const iox::units::Duration timeToWait) noexcept
+    : m_timeToWait(timeToWait)
+    , m_endTime(getCurrentMonotonicTime() + timeToWait)
+{
+}
+
+bool DeadlineTimer::hasExpired() noexcept
+{
+    return getCurrentMonotonicTime() >= m_endTime;
+}
+
+void DeadlineTimer::reset() noexcept
+{
+    m_endTime = getCurrentMonotonicTime() + m_timeToWait;
+}
+
+void DeadlineTimer::reset(const iox::units::Duration timeToWait) noexcept
+{
+    m_endTime = getCurrentMonotonicTime() + timeToWait;
```

Comment:
please update `m_timeToWait` and just call `reset()`. It's not much, but still a bit of code reuse.

Reply:
Yes I agree. Done